tests/plugins/richcombo/richcombo.js

```diff
@@ -56,5 +56,56 @@ bender.test( {
 
 		assert.areEqual( 0, combo._.listeners.length, 'Listeners array is empty.' );
 		assert.isTrue( listenersRemoved, 'All listeners are removed.' );
-	}
+	},
+
+	// (#1268)
+	'test updatestate': testUpdateState(),
+
+	// (#1268)
+	'test updatestate read only editor': testUpdateState( { readOnly: true } ),
+
+	// (#1268)
+	'test updatestate source mode': testUpdateState( { mode: 'source' } ),
+
+	// (#1268)
+	'test updatestate combo on': testUpdateState( { comboOn: true } )
 } );
+
+function testUpdateState( options ) {
+	return function() {
+		var editor = this.editor,
+			combo = editor.ui.get( 'custom_combo' ),
+			spy = sinon.spy( combo, 'setState' ),
+			expected = [ 1, CKEDITOR.TRISTATE_OFF ],
+			originalMode;
+
+		options = options || {};
+
+		if ( options.mode && options.mode !== editor.mode ) {
+			originalMode = editor.mode;
+			editor.mode = options.mode;
```

Comment:
It doesn't change the mod of the editor, it forces editor to think that mode is changed, but in reality it stays the same.